conda_build/utils.py

```diff
@@ -68,6 +68,15 @@ def tar_xf(tarball, dir_path, mode='r:*'):
         subprocess.check_call([uncompress, '-f', tarball])
         tarball = tarball[:-2]
     if not PY3 and tarball.endswith('.tar.xz'):
+        try:
+            # try extracting directly with tar,
+            # which may be able to cope with xz compression
+            tar = external.find_executable('tar')
+            subprocess.check_call([tar, '-x', '-C', dir_path, '-f', tarball])
+            return
+        except:
```

Comment:
You should only catch the exception raised by subprocess (and also guard against `tar` being `None`).